### DISCUSSION TITLE
How to explicitly set the scoring metric in `SplineCV`?

### DISCUSSION THREAD

**Discussion by JamesSample** [fatiando/community#61] [Q&A] [answered]
Thanks for Verde - I'm just getting started, but it looks like a fantastic package!

I have what is probably a very simple question, but so far I haven't figured it out: **how can I explicitly set the scoring metric used by `SplineCV`**?

At present, the [model selection example from the docs](https://www.fatiando.org/verde/latest/tutorials/model_selection.html#cross-validated-gridders) generates lots of warnings

> FutureWarning: The default scoring will change from R² to negative root mean squared error (RMSE) in Verde 2.0.0. This may change model selection results slightly.

This is helpful, but I'd like to silence these warnings and - ideally - switch to using negative root mean squared error in my current code. If I've understood correctly, it looks as though I should be able to specify any of the "regression" scoring metrics listed [here](https://scikit-learn.org/stable/modules/model_evaluation.html), e.g. by passing `scoring='neg_root_mean_squared_error'`. However, I haven't managed to get this working with `SplineCV` yet.

Using the standard `Spline`, I can can evaluate a model with the default parameters using 5-fold CV as follows

```python
spline = vd.Spline()
score_default = np.mean(
    vd.cross_val_score(spline, data_coords, data_values, scoring="neg_root_mean_squared_error")
)
print("  Neg RMSE with defaults:", score_default)
```
How can I do something similar with `SplineCV`, please?

If I do

```python
spline = vd.SplineCV(dampings=dampings, mindists=min_dists)
spline.fit(data_coords, data_values)
print("  Highest score:", spline.scores_.max())
print("  Best damping:", spline.damping_)
print("  Best mindist:", spline.mindist_)
```

It uses R2 as the default and generates lots of warnings. Is there a simple way to specify `scoring="neg_root_mean_squared_error"` in this context?

Thank you!

**Comment by leouieda** [comment, marked as answer]
Hi @JamesSample that's a great question. I actually forgot to build a way to do that into `SplineCV` when `scoring` was introduced as an argument in `cross_val_score`. I've opened fatiando/verde#379 to add this to our TODO list for the next release, which I'm hoping to get out within the next month. 

You're more than welcome to take a stab at adding this if you're interested. I laid out the changes needed in that issue and it would be a matter of changing 2 lines of code + a test. Any help we can get is always much appreciated ❤️

**Reply by JamesSample** [reply]
Thanks for the fast and helpful reply! 

I've had a go at a pull request linked to the [issue you created](https://github.com/fatiando/verde/issues/379). Fixing the code itself was very simple (as you described). I hope the test I created is OK too, but let me know if not.

I'm looking forward to getting to know Verde better!

**Reply by leouieda** [reply]
Thanks for the PR @JamesSample! Always happy to have more people on board if you're interested. And feel free to ask if you come across any issues or have any ideas.

**Comment by leouieda** [comment]
Implemented in https://github.com/fatiando/verde/pull/380 and will be available in Verde v1.8.0.
